delfin/drivers/hitachi/vsp/rest_handler.py

```diff
@@ -0,0 +1,208 @@
+                        self.rest_client.session.headers[
+                            RestHandler.HDSVSP_AUTH_KEY] = access_session
+                    else:
+                        LOG.error("Login error. URL: %(url)s\n"
+                                  "Reason: %(reason)s.",
+                                  {"url": url, "reason": res.text})
+                        if 'invalid username or password' in res.text:
+                            raise exception.InvalidUsernameOrPassword()
+                        else:
+                            raise exception.BadResponse(res.text)
+            else:
+                LOG.error('Login Parameter error')
+
+            return access_session
+        except Exception as e:
+            LOG.error("Login error: %s", six.text_type(e))
+            raise e
+
+    def logout(self):
+        try:
+            url = RestHandler.HDSVSP_LOGOUT_URL
+            if self.hdsvsp_session_id is not None:
+                url = '%s%s/sessions/%s' % \
+                      (RestHandler.HDSVSP_COMM_URL,
+                       self.storage_device_id,
+                       self.hdsvsp_session_id)
+                if self.rest_client.san_address:
+                    self.call(url, method='DELETE')
+                    self.rest_client.rest_auth_token = None
+            else:
+                LOG.error('logout error:session id not found')
+        except Exception as err:
+            LOG.error('logout error:{}'.format(err))
+            raise exception.StorageBackendException(
+                reason='Failed to Logout from restful')
+
+    def get_device_id(self):
+        try:
+            if self.rest_client.session is None:
+                self.rest_client.init_http_head()
+            storage_systems = self.get_system_info()
+            system_info = storage_systems.get('data')
+            for system in system_info:
+                if system.get('model') in consts.VSP_MODEL_NOT_USE_SVPIP:
+                    if system.get('ctl1Ip') == self.rest_client.rest_host or \
+                            system.get('ctl2Ip') == self.rest_client.rest_host:
+                        self.storage_device_id = system.get('storageDeviceId')
+                        self.device_model = system.get('model')
+                        self.serial_number = system.get('serialNumber')
+                        break
+                elif system.get('svpIp') == self.rest_client.rest_host:
+                    self.storage_device_id = system.get('storageDeviceId')
+                    self.device_model = system.get('model')
+                    self.serial_number = system.get('serialNumber')
+                    break
+            if self.storage_device_id is None:
+                LOG.error("Get device id fail,model or something is wrong")
+        except Exception as e:
+            LOG.error("Get device id error: %s", six.text_type(e))
+            raise e
+
+    def get_storage(self):
```

Comment:
The return value is `firmware_version`, so this function's name should be `get_firmware_version`?

Reply:
changed